parser_library/src/context/hlasm_context.cpp

```diff
@@ -248,12 +249,13 @@ bool hlasm_context::is_opcode(id_index symbol) const
     return macros_.find(symbol) != macros_.end() || instruction_map_.find(symbol) != instruction_map_.end();
 }
 
-hlasm_context::hlasm_context(std::string file_name)
+hlasm_context::hlasm_context(std::string file_name, std::map<std::string, std::string> asm_options)
     : instruction_map_(init_instruction_map())
     , SYSNDX_(0)
     , ord_ctx(ids_)
     , lsp_ctx(std::make_shared<lsp_context>())
 {
+    asm_options_ = asm_options;
```

Comment:
```suggestion
    asm_options_ = std::move(asm_options);
```
or better yet, move it to the initialization part.